src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageSectorSelection/ImageSectorSelection.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import {action, computed, observable} from 'mobx';
+import React from 'react';
+import RectangleSelection from '../RectangleSelection';
+import RoundingNormalizer from '../RectangleSelection/dataNormalizers/RoundingNormalizer';
+import type {SelectionData} from '../RectangleSelection/types';
+import log from 'loglevel';
+import {observer} from 'mobx-react';
+import selectionStyles from './imageSectorSelection.scss';
+
+@observer
+export default class ImageSectorSelection extends React.PureComponent {
+    props: {
+        /** Determines the position at which the selection box is rendered at the beginning. */
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        src: string,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+
+    @observable imageLoaded = false;
+    @observable containerWidth: number;
```

Comment:
The image needs to be resized with respect to the maximum available space (the container is just a div with `height: 100%` and `width: 100%` to be able to read the `clientHeight` and `clientWidth`). To make sure the image is resized correctly, we need to now if it touches top/bottom or left/right of the container, or if the image fits into the container completely. Therfore, the rendering of the image is dependent on the width of the container and the height of the container. Because these two variables cannot be derived from anything else in the component the have to be part of the state.